docs/articles/documentation/test-api/intercepting-and-mocking-http-requests.md

```diff
@@ -0,0 +1,547 @@
+
+    ```js
+    export default class RequestHook {
+        constructor (requestFilterRules, /* other params */) {
+            console.log(requestFilterRules[0]); // http://example.com
+            console.log(requestFilterRules[1]); // /\/api\/users\//
+        }
+    ```
+
+* Before sending the request, the `onRequest` method is called. Use this method to handle the request sending. If necessary, you can change the request parameters before it is sent.
+
+    This method is abstract in the base class and needs to be overriden in the descendant.
+
+    ```js
+    onRequest (/*RequestEvent event*/) {
+        throw new Error('Not implemented');
+    }
+    ```
+
+* When a response is received, the hook starts preparing to call the `onResponse` method that handles the response.
+
+    At this moment, the hook processes settings that define whether to pass the response headers and body to the `onResponse` method. These settings are specified in the second constructor parameter. This parameter takes an object with two properties - `includeHeaders` and `includeBody`.
+
+    ```js
+    export default class RequestHook {
+        constructor (requestFilterRules, responseEventConfigureOpts) {
+            console.log(responseEventConfigureOpts.includeHeaders); // false
+            console.log(responseEventConfigureOpts.includeBody);    // false
+        }
+    ```
+
+* After all the necessary preparations are done, the `onResponse` method is called. This method is abstract in the base class. Override it in the descendant to handle the request sending.
+
+    ```js
+    onResponse (/*ResponseEvent event*/) {
+        throw new Error('Not implemented');
+    }
+    ```
+
+### Writing a Hook
+
+To sum up, this is what you need to write a custom hook.
+
+* inherit from the `RequestHook` class,
+* override the `onRequest` method to handle sending the request,
+* override the `onResponse` method to handle receiving the response.
+
+```js
+import { RequestHook } from 'testcafe';
+
+class MyRequestHook extends RequestHook {
+    constructor (requestFilterRules, responseEventConfigureOpts) {
+        super(requestFilterRules, responseEventConfigureOpts);
+        // ...
+    }
+    onRequest (event) {
+        // ...
+    }
+    onResponse (event) {
+        // ...
+    }
+}
+```
+
+The `onRequest` and `onResponse` methods receive an object that contains the event parameters.
+
+The `onRequest` method's `event` object exposes the following fields.
+
+Property | Type | Description
+-------- | ---- | --------------
+`requestOptions` | Object | Contains the request parameters. You can use it to change the request parameters before the request is sent.
+`isAjax`         | Boolean | Specifies if the request is performed using AJAX.
+
+The `requestOptions` object has the following properties.
+
+Property | Type | Description
+-------- | ---- | ------------
+`headers`     | Object  | The request headers in the property-value form.
```

Comment:
These option tables contain a lot of stuff, let's add new item to the table of contents. It should be done for other huge option tables as well.